services/src/main/java/org/apache/druid/cli/CliIndexer.java

```diff
@@ -146,6 +150,30 @@ public void configure(Binder binder)
             JsonConfigProvider.bind(binder, "druid", DruidNode.class, Parent.class);
             JsonConfigProvider.bind(binder, "druid.worker", WorkerConfig.class);
 
+            String serverViewType = (String) properties.getOrDefault(
+                ServerViewModule.SERVERVIEW_TYPE_PROPERTY,
+                ServerViewModule.DEFAULT_SERVERVIEW_TYPE
+            );
+
+            if (Boolean.parseBoolean(properties.getProperty(CliCoordinator.CENTRALIZED_DATASOURCE_SCHEMA_ENABLED))
+                && !serverViewType.equals(ServerViewModule.SERVERVIEW_TYPE_HTTP)) {
+              throw DruidException
+                  .forPersona(DruidException.Persona.ADMIN)
+                  .ofCategory(DruidException.Category.UNSUPPORTED)
+                  .build(
+                      StringUtils.format(
+                          "CentralizedDatasourceSchema feature is incompatible with config %1$s=%2$s. "
+                          + "Please consider switching to http based segment discovery (set %1$s=%3$s) "
+                          + "or disable the feature (set %4$s=false).",
+                          ServerViewModule.SERVERVIEW_TYPE_PROPERTY,
+                          serverViewType,
+                          ServerViewModule.SERVERVIEW_TYPE_HTTP,
+                          CliCoordinator.CENTRALIZED_DATASOURCE_SCHEMA_ENABLED
+                      ));
+            }
+
+            JsonConfigProvider.bind(binder, "druid.centralizedDatasourceSchema", CentralizedDatasourceSchemaConfig.class);
```

Comment:
The task uses this config to build and publish schema along with segment metadata. 

In general we are ensuring that none of the services come up in case of incompatible configs.